### PR TITLE
Add Jest testing framework and an example test

### LINKED ISSUE #39: Set up automated testing
We should be writing tests as we write components.  This will be much easier to do up front than it will be to retrofit automated testing.

This issue is about setting up the test framework and incorporating it into CI, not actually writing tests for any code that has been written so far (that should be made as separate issues if it applies).

* [x] Lets use ~[mocha](https://mochajs.org/) and [chai](https://www.chaijs.com/)~ [Jest](https://jestjs.io)
* [x] Lets have travis CI run the tests

### PR DESCRIPTION
**Pull request by reefdog**
For our testing, we’ve chosen [Jest](https://jestjs.io/). It’s fast, easy to setup, and easy to use. I’ve also created a simple example test for the `HelloWorldCrawler`.

Run tests with `yarn test`, or start a watcher with `yarn test:watcher`. Create tests for individual files as `{filename}.test.js`, and put them in a `__test__` directory alongside the file they test. E.g., `/src/server/Foo.js` is tested by `/src/server/__test__/Foo.test.js`.

- [x] Add testing framework
- [x] Add Travis CI integration

Closes #39

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,3 +5,5 @@ node_js:
 
 script:
   - npm run lint
+  - npm run test
```
Comment (reefdog) [inline .travis.yml]: (Actually I should probably make a special `test-ci` that's not verbose and doesn't test for coverage.)

Comment (slifty) [reply .travis.yml]: Oh I approved and you already merged, but actually I just realized you should probably make a special `test-ci` that's not verbose and doesn't test for coverage.

Comment (reefdog) [reply .travis.yml]: np, I'll forcefully reset master and we can recreate all of this (at a bar)

Comment (slifty) [reply .travis.yml]: It'll be great to finally meet you.

Comment (reefdog) [reply .travis.yml]: It'll definitely live up to expectations!

**Pull request merged**